### PR TITLE
feat: add performance schema for Aquila

### PR DESCRIPTION
**Pull request by guomanmin**
*Issue #, if available:*

*Description of changes:*
adding new device geometry and control fields data to aquila device

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-schemas-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-schemas-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by math411** [comment]
Would it make sense to use Annotated with a Field for Decimal typing?

**Comment by math411** [comment]
As an example:

https://docs.pydantic.dev/latest/api/standard_library_types/#decimaldecimal

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 1]
```diff
@@ -126,19 +126,147 @@ class PerformanceLattice(BaseModel):
...
+    T2StarEnsemble: Decimal
+    T2EchoSingle: Decimal
+    T2EchoEnsemble: Decimal
+    T2RabiSingle: Decimal
+    T2RabiEnsemble: Decimal
+    T2BlockadedRabiSingle: Decimal
+    T2BlockadedRabiEnsemble: Decimal
+    detuningError: Decimal
+    detuningInhomogeneity: Decimal
+    rabiAmplitudeRampCorrection: List[RabiCorrection]
```
Comment (steven09200808) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:269]: Looks like we don't define List from typing. Do we need to add from typing import List here?

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 2]
```diff
@@ -126,19 +126,147 @@ class PerformanceLattice(BaseModel):
     Attributes:
         positionErrorAbs (Decimal): Error between target and actual site
```
Comment (steven09200808) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:127]: We also want to modify description here.

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 1 above)
Comment (math411) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:269]: We should also consider the built in list type

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 3]
```diff
@@ -126,19 +126,147 @@ class PerformanceLattice(BaseModel):
...
             position (measured in meters)
+        sitePositionError (Decimal): Systematic, pattern-dependent error
+            between specified and actual lattice site positions (measured in meters)
+        atomPositionError (Decimal): Random error in the atom position during
+            coherent evolution as a result of thermal motion (measured in meters)
+        fillingErrorTypical (Decimal): Typical probability of failing to occupy a site specified
+            by user as 'filled'. Upper bound that includes the pattern-dependence and site position
+            dependence. Normalized to 1.
+        fillingErrorWorst (Decimal): Worst-case probability of failing to occupy a site specified
+            by user as 'filled'. These probabilities are dependent on the pattern and site position
```
Comment (steven09200808) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:137]: Can we also modify the description of fillingErrorWorst as well?

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 1 above)
Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py]: Good catch, addressed

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 3 above)
Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:135]: Thanks for catching it. Addressed

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 2 above)
Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py]: Thanks for catching it. Addressed

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 4]
```diff
@@ -124,21 +124,150 @@ class PerformanceLattice(BaseModel):
...
+    T2StarEnsemble: Decimal
+    T2EchoSingle: Decimal
+    T2EchoEnsemble: Decimal
+    T2RabiSingle: Decimal
+    T2RabiEnsemble: Decimal
+    T2BlockadedRabiSingle: Decimal
+    T2BlockadedRabiEnsemble: Decimal
+    detuningError: Decimal
+    detuningInhomogeneity: Decimal
+    rabiAmplitudeRampCorrection: [RabiCorrection]
```
Comment (math411) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:270]:
```suggestion
    rabiAmplitudeRampCorrection: list[RabiCorrection]
```

**Review by math411 — CHANGES_REQUESTED** [review]

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 5]
```diff
@@ -124,21 +124,150 @@ class PerformanceLattice(BaseModel):
...
+            coherent and incoherent processes. (measured in s)
+        T2BlockadedRabiSingle (Decimal): Typical decoherence time of a single pair of driven blockaded qubits, as measured by a Rabi oscillation
+            protocol with variable pulse duration a maximum Rabi frequency. Includes coherent and incoherent processes. (measured in s)
+        T2BlockadedRabiEnsemble (Decimal): Decoherence time of an ensemble of pairs of driven blockaded qubits distributed over the user region
+            (different pairs do not interact with each other), as measured by a Rabi oscillation protocol with variable pulse duration at
+            maximum Rabi frequency. Includes coherent and incoherent processes. (measured in s)
+        detuningError (Decimal): Systematic error from specified value of the global detuning averaged over the user
+            region. (measured in rad/s)
+        detuningInhomogeneity (Decimal): RMS inhomogeneity of the detuning over the user region. (measured in rad/s)
+        rabiAmplitudeRampCorrection (List[RabiCorrection]): dynamic correction curve of effective single-qubit
```
Comment (tachikoma-li) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:246]:
```suggestion
        rabiAmplitudeRampCorrection (list[RabiCorrection]): dynamic correction curve of effective single-qubit
```
Sorry for randomly jumping into the review. It might be good to be consistent with the comment below about using built-in type.

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 4 above)
Comment (steven09200808) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:270]: I found we already majorly used List in this repository: example https://github.com/amazon-braket/amazon-braket-schemas-python/blob/2d38f86867f95194196134e463c71d340a7d8c7a/src/braket/ir/openqasm/modifiers.py#L37. So looks like List is also good here?

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 5 above)
Comment (math411) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:246]: Agreed. I will be opening a separate PR to make use of these types and for the type being, my suggestion will be edited to use the typing import.

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 4 above)
Comment (math411) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:270]: I have edit the suggestion to go back to using `List` from typing and will open a separate PR that addresses this change for the whole repo. @guomanmin , you may still need to change [L15](https://github.com/amazon-braket/amazon-braket-schemas-python/pull/167/files/50863035e1f0a32f91e0e80773a26aec2d8ef94b#diff-ecdd6f26bdf146078b4f4dbcc54c0f81e04b8f9d0e143d37243915def886eadeR15) to:
```
from typing import List, Tuple
```

**Comment by guomanmin** [comment]
> Would it make sense to use Annotated with a Field for Decimal typing?

That's an interesting concept, but I think creating a custom serializer instead of creating a new class (RabiCorrection) might be over-complicating things for rabiAmplitudeRampCorrection, if that's what you're referring to.

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 4 above)
Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py]: Reverted back to using List and imported List from typing

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 5 above)
Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py]: @tachikoma-li not random at all, thanks for your suggestion. Will default to using List from typing for the scope of this PR.

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 6]
```diff
@@ -124,21 +124,181 @@ class PerformanceLattice(BaseModel):
...
+            pattern-dependence and site position dependence. Normalized to 1.
+        vacancyErrorTypical (Annotated[Decimal, Field(ge=0, le=1)]): Typical probability of
+            erroneously filling a site specified by user as 'unfilled'. These probabilities can be
+            dependent on the pattern and site position within the pattern, and can change slightly
+            with time. Normalized to 1.
+        vacancyErrorWorst (Annotated[Decimal, Field(ge=0, le=1)]): Worst-case probability of
+            erroneously filling a site specified by user as 'unfilled'. Upper bound that includes
+            the pattern-dependence, site position dependence and time-variation of this
+            probability. Normalized to 1.
+        atomLossProbabilityTypical (Annotated[Decimal, Field(ge=0, le=1)]): Typical Probability of
```
Comment (math411) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:148]: Should the P here be capitalized?

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 7]
```diff
@@ -124,21 +124,181 @@ class PerformanceLattice(BaseModel):
...
+            atom loss from a filled site between the first and second image. These probabilities
+            can be dependent on the pattern and site position within the pattern, and can change
+            slightly with time. Normalized to 1.
+        atomLossProbabilityWorst (Annotated[Decimal, Field(ge=0, le=1)]): Worst-case probability of
+            atom loss from a filled site between the first and second image. Upper bound that
+            includes the pattern-dependence, site position dependence and time-variation of this
+            probability. Normalized to 1.
+        atomCaptureProbabilityTypical (Annotated[Decimal, Field(ge=0, le=1)]): Typical probability
+            of atom capture into an empty site between the first and second image.These
```
Comment (math411) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:157]:
```suggestion
            of atom capture into an empty site between the first and second image. These
```

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 8]
```diff
@@ -124,21 +124,181 @@ class PerformanceLattice(BaseModel):
...
+    rabiCorrection: Annotated[Decimal, Field(ge=0.0, le=1.0)]
 
 
 class PerformanceRydbergGlobal(BaseModel):
     """
     Parameters determining the limitations of the global driving field
     Attributes:
-        rabiFrequencyErrorRel (Decimal): random error in the Rabi frequency, relative (unitless)
+        rabiFrequencyErrorRel (Decimal): Total error in the Rabi frequency due to inhomogeneity and
+            variations in time, relative. (unitless)
```
Comment (math411) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:218]: What does relative mean here?

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 9]
```diff
@@ -124,21 +124,181 @@ class PerformanceLattice(BaseModel):
...
+        rabiFrequencyGlobalErrorRel (Decimal): RMS Rabi frequency variation in time as a relative
+            value. (unitless)
+        rabiFrequencyInhomogeneityRel (Decimal): RMS Rabi frequency inhomogeneity over the user
+            region, relative. (unitless)
+        groundDetectionError (Decimal): Probability of mis-detecting a ground-state atom as a
+            Rydberg-state atom. (unitless)
+        rydbergDetectionError (Decimal): Probability of mis-detecting a Rydberg-state atom as a
+            ground-state atom. (unitless)
+        groundPrepError (Annotated[Decimal, Field(ge=0, le=1)]): Probability of failing to
+            initialize an atom in the ground state
```
Comment (math411) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:228]: formatting issue here

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 10]
```diff
@@ -124,21 +124,181 @@ class PerformanceLattice(BaseModel):
...
+            prior to user-programmed coherent evolution, in the absense of any local detuning
+            pattern. Normalized to 1.
```
Comment (virajvchaudhari) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:230]: Needs proper formatting?

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 7 above)
Comment (virajvchaudhari) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:157]:
```suggestion
            of atom capture into an empty site between the first and second image. These
```

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 11]
```diff
@@ -124,21 +124,181 @@ class PerformanceLattice(BaseModel):
...
+        rydbergPrepErrorBest (Annotated[Decimal, Field(ge=0, le=1)]): Probability of failing to
+            initialize an atom in the Rydberg state by an optimal (for that site) user specified
+            pi-pulse from the ground state at maximum Rabi frequency, in the absense of any local
```
Comment (virajvchaudhari) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:233]:
```suggestion
            pi-pulse from the ground state at maximum Rabi frequency, in the absence of any local
```

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 12]
```diff
@@ -124,21 +124,181 @@ class PerformanceLattice(BaseModel):
...
+            detuning pattern. Normalized to 1.
+        rydbergPrepErrorWorst (Annotated[Decimal, Field(ge=0, le=1)]): Worst-case probability of
+            failing to initialize an atom in the Rydberg state by a user specified pi-pulse from
+            the ground state at maximum Rabi frequency, optimized for a different site, in the
+            absense of any local detuning pattern. Normalized to 1.
```
Comment (virajvchaudhari) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:238]: nit: This `absence` needs to corrected in multiple places?

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 6 above)
Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:148]: No it should not! Addressed

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 7 above)
Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:157]: Good catch. Addressed

Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:157]: Good catch. Addressed

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 11 above)
Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:233]: This inspired me to copy this whole session into google doc for a spell check, thanks

Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:233]: @maolinml a note for both of us (I think) lol

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 12 above)
Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:238]: Yep, addressed

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 9 above)
Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:228]: I am not sure if this is a formatting issue - this is done on purpose to satisfy linter's <100 character issue. Is there another way to do so?

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 10 above)
Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:230]: Closing this thread as there is another thread on the same issue

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 8 above)
Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:218]: @maolinml I think this is potentially a request for rewording. Could you chime in?

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 13]
```diff
@@ -124,21 +124,181 @@ class PerformanceLattice(BaseModel):
...
     """
     Parameters determining the limitations of the global driving field
     Attributes:
-        rabiFrequencyErrorRel (Decimal): random error in the Rabi frequency, relative (unitless)
+        rabiFrequencyErrorRel (Decimal): Total error in the Rabi frequency due to inhomogeneity and
+            variations in time, relative. (unitless)
+        rabiFrequencyGlobalErrorRel (Decimal): RMS Rabi frequency variation in time as a relative
+            value. (unitless)
+        rabiFrequencyInhomogeneityRel (Decimal): RMS Rabi frequency inhomogeneity over the user
+            region, relative. (unitless)
```
Comment (virajvchaudhari) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:222]: unitless? What does that mean? Should be elaborating what does unitless means? @maolinml @guomanmin

Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:222]: Similarly, this a request for rewording @maolinml Maybe we could re this as "This value is unitless" or just removed (unitless)?

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 14]
```diff
@@ -124,21 +124,180 @@ class PerformanceLattice(BaseModel):
...
+        groundDetectionError (Decimal): Probability of mis-detecting a ground-state atom as a
```
Comment (maolinml) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:223]: `groundDetectionError` and `rydbergDetectionError` should have the same type as, for example, `groundPrepError` because these three quantities are all probabilities which are between 0 and 1 by definition.

Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py]: addressed

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 15]
```diff
@@ -124,21 +124,180 @@ class PerformanceLattice(BaseModel):
...
+            probabilities can be dependent on the pattern and site position within the pattern, and
+            can change slightly with time. Normalized to 1.
+        atomCaptureProbabilityWorst (Annotated[Decimal, Field(ge=0, le=1)]): Worst-case probability
+            of atom capture from a filled site between the first and second image. Upper bound that
+            includes the pattern-dependence, site position dependence and time-variation of this
+            probability. Normalized to 1.
+        atomDetectionErrorFalsePositiveTypical (Annotated[Decimal, Field(ge=0, le=1)]): Typical
+            probability of a false-positive atom detection error. These probabilities can be
+            dependent on the pattern and site position within the pattern, and can change slightly
+    with time. Normalized to 1.
```
Comment (math411) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:167]: formatting issue here

Comment (virajvchaudhari) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:167]: Do you know why the linters are not catching this? The check code format passes here.

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py [hunk 16]
```diff
@@ -124,21 +124,180 @@ class PerformanceLattice(BaseModel):
...
+            pi-wait-pi protocol. (measured in s)
+        T2StarSingle (Decimal): Typical dephasing time of a single non-interacting qubit in the
+            absence of drive, as measured by a Ramsey protocol. Includes coherent and incoherent
+            processes. (measured in s)
+        T2StarEnsemble (Decimal): Dephasing time of an ensemble of non-interacting qubits
+            distributed over the user region, in the absence of drive, as measured by a Ramsey
+            protocol. Includes coherent and incoherent processes. (measured in s)
+        T2EchoSingle (Decimal): Typical dephasing time of a single non-interacting qubit in the
+            absence of drive, as measured by a spin-echo dynamical decoupling protocol. This
+            measurement isolates the effects of incoherent processes. (measured in s)
```
Comment (virajvchaudhari) [inline src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:251]: Does `measured in s` represents `seconds` here?  And can you confirm if the terms like unitless, s, will be elaborated in another PR or you are addressing those now? Other than that looks good.

**Review by virajvchaudhari — APPROVED** [review]

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 15 above)
Comment (math411) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py]: We do not verify the RST format with the current docstring linters

File: src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py (hunk 16 above)
Comment (maolinml) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:251]: 1. Yes, `measured in s` means the quantity has unit seconds.
2. Yes, units like `s`, `(unitless)`, `meters` are address as part of this PR, and there will no other PR
3. Lishuai and I have discussed on the choice of wordings for the units in his CR, and we agree to use these units.

Comment (guomanmin) [reply src/braket/device_schema/quera/quera_ahs_paradigm_properties_v1.py:251]: Yes. Using "s" instead of "seconds"  was a deliberate decision to be abbreviated, I believe - see https://code.amazon.com/reviews/CR-101330423/revisions/2#/diff

@maolinml I understand that Quera is looking to update some field descriptions in another iteration in the future. For fields like "unitless", I will lean on you as you're close to customer voice to decide if we want to update those fields - if needed, I can publish another PR and update these fields accordingly
